_includes/comments-providers/discourse.html

```diff
@@ -1,7 +1,7 @@
 {% if site.comments.discourse.server %}
-{% capture canonical %}{% if site.permalink contains '.html' %}{{ page.url | absolute_url }}{% else %}{{ page.url | absolute_url | remove:'index.html' | strip_slash }}{% endif %}{% endcapture %}
+{% capture canonical %}{{ page.url | replace: 'index.html','' | prepend: site.baseurl | prepend: site.url }}{% endcapture %}
```

Comment:
Maybe I misconfigured something, but I am pretty sure the `canonical` resulted in a relative url, and that's why I got here. Go figure.
So would this be the simpler solution?:
```
{{ page.url | replace: 'index.html','' | absoulte_url }}
```